app/orchestrators/CorrectReturnOrchestrator.scala

```diff
@@ -20,18 +20,76 @@ import cats.data.EitherT
 import com.google.inject.{Inject, Singleton}
 import connectors.SoftDrinksIndustryLevyConnector
 import errors.{FailedToAddDataToUserAnswers, NoSdilReturnForPeriod, NoVariableReturns}
-import models.backend.RetrievedSubscription
+import handlers.ErrorHandler
+import models.backend.{RetrievedSubscription, Site, UkAddress}
 import models.correctReturn.CorrectReturnUserAnswersData
+import models.enums.SiteTypes.{PRODUCTION_SITE, WAREHOUSE}
+import models.requests.DataRequest
+import models.submission.VariationsSites.getHighestRefNumber
+import models.submission.{ClosedSite, ReturnVariationData, VariationsPersonalDetails, VariationsSite, VariationsSites, VariationsSubmission}
+import models.updateRegisteredDetails.ContactDetails
 import models.{ReturnPeriod, SdilReturn, UserAnswers}
+import pages.correctReturn.{CorrectionReasonPage, RepaymentMethodPage}
+import play.api.mvc.AnyContent
+import play.api.mvc.Results.InternalServerError
 import service.VariationResult
 import services.SessionService
 import uk.gov.hmrc.http.HeaderCarrier
+import utilities.GenericLogger
 
+import java.time.Instant
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class CorrectReturnOrchestrator @Inject()(connector: SoftDrinksIndustryLevyConnector,
-                                          sessionService: SessionService){
+                                          sessionService: SessionService,
+                                          genericLogger: GenericLogger,
+                                          val errorHandler: ErrorHandler){
+
+  def submitVariation()
```

Comment:
I think we should write this so we call it as `def submitVariations(userAnswers: UserAnswers, subscription: RetrievedSubscription` rather passing in the request implicitly.
(I'm happy to take this bit over it you like - up to you if you want me to?)